src/DocBlock/Annotation.php

```diff
@@ -16,10 +16,46 @@
  * This represents an entire annotation from a docblock.
  *
  * @author Graham Campbell <graham@alt-three.com>
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
  */
 class Annotation
 {
     /**
+     * Regex to match any types, shall be used with `x` modifier.
+     *
+     * @internal
+     */
+    const REGEX_TYPES = '
+    # <scalar> is any non-array, non-generic, non-alternated type, eg `int` or `\Foo`
```

Comment:
I think you are confusing scalar and primitive:

>  PHP supports ten primitive types.
>
> - Four scalar types: boolean, integer, float (floating-point number, aka double), string.
> - Four compound types: array, object, callable, iterable.
> - And finally two special types: resource, NULL.

http://php.net/manual/en/language.types.intro.php